### DISCUSSION TITLE
Budibase with ADFS via OIDC

### DISCUSSION THREAD

**Discussion by asampels** [Budibase/budibase#8564] [Help]
So the first question is has anybody been able to setup the budibase authentication using OIDC and an ADFS server?

I think I am close, but missing something.

I have been able to setup the authentication to hairpin through keycloak then to ADFS but feel that's an extra step that is not needed.

I have been able to setup the OIDC authentication to ADFS and get the login page, but once the user is authenticated I get the following.

in the browser "{"message":"Unauthorized","status":403}"

and on the budibase console the following;

bbworker              | Authentication error InternalOAuthError: failed to fetch user profile
bbworker              |     at /app/node_modules/@techpass/passport-openidconnect/lib/strategy.js:270:23
bbworker              |     at passBackControl (/app/node_modules/oauth/lib/oauth2.js:132:9)
bbworker              |     at IncomingMessage.<anonymous> (/app/node_modules/oauth/lib/oauth2.js:157:7)
bbworker              |     at IncomingMessage.emit (events.js:412:35)
bbworker              |     at IncomingMessage.emit (domain.js:475:12)
bbworker              |     at endReadableNT (internal/streams/readable.js:1333:12)
bbworker              |     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
bbworker              |   oauthError: { statusCode: 401, data: '' }
bbworker              | }
bbworker              | ERROR [1667832940298] (29 on 4debd4278803): Unauthorized
bbworker              |     ForbiddenError: Unauthorized

It looks like I am missing information in the returned authorization "oauthError: { statusCode: 401, data: '' }" ?

Any help or direction would be appreciated.

Thanks.

**Comment by dkidd255** [comment]
Same exact issue here as well with ADFS. Anyone have any ideas?

**Comment by dkidd255** [comment]
I ended up using keycloak with no issues based on @asampels comment, there is obviously something not jiving with ADFS.
